src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -157,7 +157,7 @@ static CellIdx getLeastUsedCell(Random &rng, UInt column, //TODO remove static m
   UInt32 minNumSegments = UINT_MAX;
   UInt32 numTiedCells = 0;
   for (CellIdx cell = start; cell < end; cell++) {
-    const UInt32 numSegments = connections.numSegments(cell);
+    const UInt32 numSegments = static_cast<UInt32>(connections.numSegments(static_cast<CellIdx>(cell)));
```

Comment:
cell is a CellIdx

Reply:
> cell is a CellIdx

Ok, I can fix that...and a few other lines around it seem to also need to be CellIdx.

Reply:
I mean just 
- static_cast<CellIdx>(cell))
+ cell